Makefile

```diff
@@ -1,11 +1,31 @@
 CXX=g++
 CXXOPTIMIZE= -O2
-CXXFLAGS= -g -Wall -pthread -std=c++11 $(CXXOPTIMIZE)
-SRCFILES = main.cpp server.cpp reply.cpp config.cc config_parser.cc
+CXXFLAGS= -g -Wall -pthread -std=c++11
+SRCFILES = server.cpp reply.cpp config.cc config_parser.cc
+GTEST_DIR = googletest/googletest
+
+all: CXXFLAGS += $(CXXOPTIMIZE)
+all: SRCFILES += main.cpp
+all: webserver
+
+test: CXXFLAGS += -isystem ${GTEST_DIR}/include
+test: SRCFILES += config_parser_test.cc config_test.cc ${GTEST_DIR}/src/gtest_main.cc libgtest.a
+test: | clean webserver_test
+
+run:
+	./webserver
+
+run_coverage:
+	./webserver_test
+	gcov -r server.cpp reply.cpp config.cc config_parser.cc
```

Comment:
Great additions to our Makefile! Now we can our tests and see if it covers everything from the classes and files that we have in this repository.